src/app/shared/components/List/ListItem.tsx

```diff
@@ -199,12 +205,14 @@ export default function ListItem<T extends { id: string }>({
 
             function handleOpenPosition() {
               const element = menuButtonRef.current;
+              const contextMenu = menuItemsRef?.current?.offsetHeight || 300;
               if (!element) return;
+              if (!contextMenu) return;
 
               const { bottom } = element.getBoundingClientRect();
               const windowHeight = window.innerHeight;
 
-              const isHalfway = bottom > windowHeight / 2;
+              const isHalfway = bottom + contextMenu > windowHeight;
               setIsHalfwayDown(isHalfway);
```

Comment:
I think this no longer represents whether it is below the middle of the screen or not, better to update the state name as well to avoid confusion.
Now it seems that it always opens downwards as long as it fits, no?